src/panels/config/js/script/device.tsx

```diff
@@ -48,16 +65,77 @@ export default class DeviceActionEditor extends Component<
           hass={hass}
           label="Action"
         />
+        {extraFieldsData && (
+          <ha-form
+            data={Object.assign({}, ...extraFieldsData)}
+            onData-changed={this._extraFieldsChanged}
+            schema={this.state.capabilities.extra_fields}
+            computeLabel={this._extraFieldsComputeLabelCallback(hass.localize)}
+          />
+        )}
       </div>
     );
   }
 
+  public componentDidMount() {
+    if (!this.state.capabilities) {
+      this._getCapabilities();
+    }
+    if (this.props.action) {
+      this._origAction = this.props.action;
+    }
+  }
+
+  public componentDidUpdate(prevProps) {
+    if (prevProps.action !== this.props.action) {
```

Comment:
We should use the `deviceAutomationsEqual` function here (also on the others btw)